src/factory/load_gltf.rs

```diff
@@ -182,239 +258,221 @@ impl super::Factory {
-                let sampler = channel.sampler();
-                let target = channel.target();
-                let node = target.node();
-                let object = match node_map.get(&node.index()) {
-                    Some(object) => object.clone(),
-                    // This animation does not correspond to any loaded node.
-                    None => continue,
-                };
-                let input = sampler.input();
-                let output = sampler.output();
-                let interpolation = match sampler.interpolation() {
-                    Linear => animation::Interpolation::Linear,
-                    Step => animation::Interpolation::Discrete,
-                    CubicSpline => animation::Interpolation::Cubic,
-                    CatmullRomSpline => animation::Interpolation::CatmullRom,
-                };
-                use animation::{Binding, Track, Values};
-                let times: Vec<f32> = AccessorIter::new(input, buffers).collect();
-                let (binding, values) = match target.path() {
-                    gltf::animation::TrsProperty::Translation => {
-                        let values = AccessorIter::<[f32; 3]>::new(output, buffers)
-                            .map(|v| mint::Vector3::from(v))
-                            .collect::<Vec<_>>();
-                        assert_eq!(values.len(), times.len());
-                        (Binding::Position, Values::Vector3(values))
-                    }
-                    gltf::animation::TrsProperty::Rotation => {
-                        let values = AccessorIter::<[f32; 4]>::new(output, buffers)
-                            .map(|r| mint::Quaternion::from(r))
-                            .collect::<Vec<_>>();
-                        assert_eq!(values.len(), times.len());
-                        (Binding::Orientation, Values::Quaternion(values))
-                    }
-                    gltf::animation::TrsProperty::Scale => {
-                        // TODO: Groups do not handle non-uniform scaling, so for now
-                        // we'll choose Y to be the scale factor in all directions.
-                        let values = AccessorIter::<[f32; 3]>::new(output, buffers)
-                            .map(|s| s[1])
-                            .collect::<Vec<_>>();
-                        assert_eq!(values.len(), times.len());
-                        (Binding::Scale, Values::Scalar(values))
+fn bind_objects(
+    factory: &mut Factory,
+    gltf: &gltf::Gltf,
+    heirarchy: &VecMap<Group>,
+    meshes: &VecMap<Vec<Mesh>>,
+    skeletons: &[Skeleton],
+) -> Vec<Mesh> {
+    let mut instances = Vec::new();
+    for node in gltf.nodes() {
+        if let Some(ref group) = heirarchy.get(node.index()) {
+            if let Some(mesh) = node.mesh() {
+                let mut primitives = meshes[mesh.index()]
+                    .iter()
+                    .map(|template| factory.mesh_instance(template))
```

Comment:
@vitvakatu How does this align with the new auto-instancing?

Reply:
Well, if meshes have the same material - they will be auto-grouped and rendered as instanced.